packages/astro/src/runtime/server/index.ts

```diff
@@ -263,6 +263,10 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 
 /** Create the Astro.fetchContent() runtime function. */
 function createFetchContentFn(url: URL, site: URL) {
+	let sitePathname = site.pathname;
+	if(!sitePathname.endsWith('/')) {
```

Comment:
I made a change in #2471 so that this is always true (`site` always ends with `/`). Check is no longer needed!

Reply:
I'll test again, but this was indeed causing an issue.

Reply:
For reference, this is the line that should make that moot: https://github.com/withastro/astro/blob/c9bb1147cbfae20e3ecdf29ef2866a183b3b18e3/packages/astro/src/core/config.ts#L59

Reply:
It's because your tests add their own config that doesn't go through this schema stuff. I'll see if I can fix that.

Reply:
Ah, I wonder if we can make the tests go through the "user config -> final config" pipeline, instead of just submitting their own final config. A PR for another time, I guess

Reply:
Yes, I thought about doing this, but worried it might break a bunch of tests. I might try in another branch just to see.